src/Raven.Server/Documents/Indexes/Persistence/Lucene/OutputReduceIndexWriteOperation.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Raven.Client.Data.Indexes;
+using Raven.NewClient.Client.Blittable;
+using Raven.NewClient.Client.Document;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
+using Raven.Server.Documents.Indexes.Persistence.Lucene.Documents;
+using Raven.Server.Indexing;
+using Raven.Server.NotificationCenter.Notifications;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Voron.Impl;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Lucene
+{
+    public class OutputReduceIndexWriteOperation : IndexWriteOperation
+    {
+        private readonly OutputReduceToCollectionCommand _outputReduceToCollectionCommand;
+
+        public OutputReduceIndexWriteOperation(MapReduceIndex index, LuceneVoronDirectory directory, LuceneDocumentConverterBase converter, Transaction writeTransaction, LuceneIndexPersistence persistence) 
+            : base(index, directory, converter, writeTransaction, persistence)
+        {
+            var outputReduceToCollection = index.Definition.OutputReduceToCollection;
+            Debug.Assert(string.IsNullOrWhiteSpace(outputReduceToCollection) == false);
+            _outputReduceToCollectionCommand = new OutputReduceToCollectionCommand(DocumentDatabase, outputReduceToCollection);
+        }
+
+        public override void Commit(IndexingStatsScope stats)
+        {
+            using (stats.For(IndexingOperation.Reduce.SaveOutputDocuments))
+            {
+                var enqueue = DocumentDatabase.TxMerger.Enqueue(_outputReduceToCollectionCommand);
+                base.Commit(stats);
```

Comment:
It will be handled as the higher level in indexing as before. 
Note that we had try..finally around this, not try..catch.